Improve open source projects support.

haydenroche5 opened the pull request:
- Add a wolfssl_%.bbappend for curl.
- Add curl_7.82.0.bbappend.
- Add support for OpenSSH 8.5p1.
- Improve documentation in README.md around building open source projects with wolfSSL.
- Clean up long lines and other minor things in README.md.

cconlon commented on README.md:
```diff
@@ -12,23 +13,18 @@ This layer currently provides recipes for the following wolfSSL products:
-meta-networking layer, located [here](
-https://github.com/openembedded/meta-openembedded/blob/master/meta-networking/recipes-connectivity/wolfssl/wolfssl_3.14.4.bb).
+The wolfSSL library recipe is also included in the openembedded meta-networking
+layer, located [here](https://github.com/openembedded/meta-openembedded/blob/master/meta-networking/recipes-connectivity/wolfssl/wolfssl_3.14.4.bb).
```
As long as we're updating this, can we make this link more generic - pointing to "https://github.com/openembedded/meta-openembedded/blob/master/meta-networking/recipes-connectivity/wolfssl", rather than the specific old 3.14.4?

cconlon commented on README.md:
```diff
@@ -152,20 +181,22 @@ terminal.
-and a standard commercial license. For commercial license
-questions, please contact wolfSSL at licensing@wolfssl.com. For product
-support inquiries please contact support@wolfssl.com.
+wolfSSL is open source and dual licensed under both the GPLv2 and a standard
```
Unrelated to your changes here, but let's add a note about the availability of our FIPS validated wolfCrypt in this license section, with a link to https://www.wolfssl.com/license/fips/.

cconlon commented on recipes-connectivity/openssh/files/wolfssl-openssh-8.5p1.patch:
```diff
@@ -0,0 +1,531 @@
+ 
+ 	switch (key->type) {
+-#ifdef WITH_OPENSSL
++#if defined(WITH_OPENSSL)
```
Are these switches from `#ifdef` to `#if defined` necessary?  Also one below this in authfile.c.

cconlon commented on recipes-connectivity/openssh/files/wolfssl-openssh-8.5p1.patch:
```diff
@@ -0,0 +1,531 @@
+ 		       return SSH_ERR_LIBCRYPTO_ERROR;
+ 	} else if (!EVP_CIPHER_CTX_get_iv(cc->evp, iv, len))
+-	       return SSH_ERR_LIBCRYPTO_ERROR;
++               return SSH_ERR_LIBCRYPTO_ERROR;
```
Looks like this can be removed, just whitespace difference.

cconlon commented on recipes-connectivity/openssh/files/wolfssl-openssh-8.5p1.patch:
```diff
@@ -0,0 +1,531 @@
+ 
++static void Logging_cb(const int logLevel, const char *const logMessage) {
++	// FILE* f;
++	// f = fopen("/home/hayden/lenovo/openssh-8.5p1/wolf.log", "a");
```
Do you want to leave this in here?  OK with me, so we know where to add it back in if needed, but might want to make your path more generic?

haydenroche5 commented:
@cconlon addressed your feedback and pushed the updated patch to OSP [here](https://github.com/wolfSSL/osp/pull/111). Once that one's merged, I'll incorporate the OpenSSH patch in this PR.

cconlon commented:
@haydenroche5 thanks, OSP PR has been merged in.
